### ISSUE TITLE
Document how to debug/profile spark hellbender tools

### ISSUE THREAD

**Issue by droazen** [broadinstitute/gatk#972]

**Comment by droazen** [comment]
@laserson or @tomwhite if you could give us some pointers to best-practice docs here, it would be much appreciated!

**Comment by laserson** [comment]
cc @sryza for any pointers.  I'll start with Sandy's excellent two blog posts on tuning Spark jobs if you haven't seen them already:
http://blog.cloudera.com/blog/2015/03/how-to-tune-your-apache-spark-jobs-part-1/
http://blog.cloudera.com/blog/2015/03/how-to-tune-your-apache-spark-jobs-part-2/

**Comment by tomwhite** [comment]
In addition to the tuning links that @laserson referenced I would add that the Spark UI has a lot of information about the running job (DAG showing number of stages, amount of data that is sent through each shuffle, etc). I've opened https://github.com/broadinstitute/gatk/pull/1193 to make it easier to debug Spark tests using the UI.

I've also had some success with using visualvm to profile a Spark job (running locally). This points to hotspots which can help indicate where the job is spending its time. (E.g. I saw a job was spending a lot of time doing serialization which indicated that we shouldn't be using SAMRecord as the key.)
